eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -2166,16 +2188,52 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
     skip.foreach(tx => log.info(s"no need to watch txid=${tx.txid}, it has already been confirmed"))
   }
 
-  def doPublish(localCommitPublished: LocalCommitPublished): Unit = {
+  def doPublish(localCommitPublished: LocalCommitPublished, commitments: Commitments): Unit = {
```

Comment:
That means `Helpers` should return a set of `PublishAsap` objects?
I'll have a look at that.
Do note that this will be refactored in later PRs, my goal is to indeed move most of this logic out of here, but I wanted to get a first result with a reasonable amount of new code.

Reply:
Done in https://github.com/ACINQ/eclair/pull/1702/commits/5d88187901aa9a1892726540ff68d82dc77a7961